subsys/bluetooth/mesh/dfu_cli.c

```diff
@@ -963,17 +963,22 @@ const struct bt_mesh_model_op _bt_mesh_dfu_cli_op[] = {
 
 static int dfu_cli_init(const struct bt_mesh_model *mod)
 {
+	int err;
 	struct bt_mesh_dfu_cli *cli = mod->rt->user_data;
+	cli->mod = mod;
+
+	const struct bt_mesh_model *blob_cli =
+		bt_mesh_model_find(bt_mesh_model_elem(mod), BT_MESH_MODEL_ID_BLOB_CLI);
 
-	if (mod->rt->elem_idx != 0) {
-		LOG_ERR("DFU update client must be instantiated on first elem");
+	if (blob_cli == NULL) {
+		LOG_ERR("Firmware Update Client cannot extend BLOB Transfer Client");
```

Comment:
```suggestion
		LOG_ERR("BLOB Transfer Client needed");
```

or 

```suggestion
		LOG_ERR("Missing BLOB cli");
```


Long debug string are not very useful and eat up the flash during development stage leading to misconceptions about footprint.